api/ecosystem_test.go

```diff
@@ -0,0 +1,238 @@
+	// this test.
+	stLeader, err := createServerTester("TestHostPoorConnectivity - Leader")
+	if err != nil {
+		t.Fatal(err)
+	}
+	stHost1, err := blankServerTester("TestHostPoorConnectivity - Host 1")
+	if err != nil {
+		t.Fatal(err)
+	}
+	stHost2, err := blankServerTester("TestHostPoorConnectivity - Host 2")
+	if err != nil {
+		t.Fatal(err)
+	}
+	stHost3, err := blankServerTester("TestHostPoorConnectivity - Host 3")
+	if err != nil {
+		t.Fatal(err)
+	}
+	stHost4, err := blankServerTester("TestHostPoorConnectivity - Host 4")
+	if err != nil {
+		t.Fatal(err)
+	}
+	stRenter1, err := blankServerTester("TestHostPoorConnectivity - Renter 1")
+	if err != nil {
+		t.Fatal(err)
+	}
+	stRenter2, err := blankServerTester("TestHostPoorConnectivity - Renter 2")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Fetch all of the addresses of the nodes that got created.
+	var ggSTL, ggSTH1, ggSTH2, ggSTH3, ggSTH4, ggSTR1, ggSTR2 GatewayGET
+	err = stLeader.getAPI("/gateway", &ggSTL)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = stHost1.getAPI("/gateway", &ggSTH1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = stHost2.getAPI("/gateway", &ggSTH2)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = stHost3.getAPI("/gateway", &ggSTH3)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = stHost4.getAPI("/gateway", &ggSTH4)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = stRenter1.getAPI("/gateway", &ggSTR1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = stRenter2.getAPI("/gateway", &ggSTR2)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Connect all of the peers in a circle, so that everyone is connected but
+	// there are a lot of hops.
+	err = stLeader.stdPostAPI("/gateway/connect/"+string(ggSTH1.NetAddress), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = stHost1.stdPostAPI("/gateway/connect/"+string(ggSTH2.NetAddress), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = stHost2.stdPostAPI("/gateway/connect/"+string(ggSTH3.NetAddress), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = stHost3.stdPostAPI("/gateway/connect/"+string(ggSTH4.NetAddress), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = stHost4.stdPostAPI("/gateway/connect/"+string(ggSTR1.NetAddress), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = stRenter1.stdPostAPI("/gateway/connect/"+string(ggSTR2.NetAddress), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = stRenter2.stdPostAPI("/gateway/connect/"+string(ggSTL.NetAddress), nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Connectivity check - all nodes should by synchronized to the leader's
+	// chain, which should have been the longest.
+	allTesters := []*serverTester{stLeader, stHost1, stHost2, stHost3, stHost4, stRenter1, stRenter2}
+	chainTip, err := synchronizationCheck(allTesters)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Mine a block from each node, to give the node money in the wallet that
+	// is recognized by the shared chain.
+	for i := range allTesters {
+		// Wait until the current tester has 'leadBlockID' as its current
```

Comment:
probably clearer to just call it `chainTip` instead of 'leadBlockID' since the latter isn't used anywhere else